File: DocTok.Shared/ResponseModels/Document/Detail/DocumentDetailResponseModel.cs
using DocTok.Shared.ResponseModels.Document.Short;

namespace DocTok.Shared.ResponseModels.Document.Detail
{
    public class DocumentDetailResponseModel
    {
        public int Id { get; set; }

        public int? ParentId { get; set; }

        public string Caption { get; set; }

        public string Content { get; set; }

        public int ProjectId { get; set; } = 1;

        public List<DocumentHierarсhyModel> Childs { get; set; } = new();

        public int CreatedBy { get; set; }
    }
}

File: DocTok/ClientApp/src/models/responseModels/Detail/DocumentDetailResponseModel.ts
import {DocumentHierarchyModel}  from "../DocumentHierarchyModel";

export type DocumentPageDetailResponseModel ={
    id: number,
    parentId: number,
    caption: string,
    content: string,
    projectId: number,
    childs: DocumentHierarchyModel[],
    createdBy: number,
}